### PR TITLE
[DOC] transfer dalapi

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

- Set up dalapi for daal repo
- Test on array.hpp + temporary array.rst file

Do not merge.

### CODE REVIEW

**Comment by rlnx** [comment]
There are some files in `__pycache__` folder, add them to `.gitignore`

**Comment by rlnx** [comment]
`docs/include` should be removed and dalapi configured to parse headers in `cpp/oneapi/dal`.

**Comment by outoftardis** [comment]
> `docs/include` should be removed and dalapi configured to parse headers in `cpp/oneapi/dal`.

I'm aware. This is just for test purposes.

File: cpp/oneapi/dal/array.hpp [hunk 1]
```diff
@@ -24,6 +24,10 @@
 namespace oneapi::dal {
 namespace v1 {
 
+/// @tparam T The type of the memory block elements within the array.
+///              $T$ can represent any type.
```
Comment (rlnx) [inline cpp/oneapi/dal/array.hpp:28]: `$...$` is used for mathematical expressions, to refer to the template parameter or normal parameters of the function, use ``:literal:`...` ``. It should be changed in the spec as well

**Pull request closed without merging**